presentation/profile/src/main/java/com/giraffe/profile/screens/profile/SettingsViewModel.kt

```diff
@@ -0,0 +1,140 @@
+package com.giraffe.profile.screens.profile
+
+import androidx.lifecycle.viewModelScope
+import com.giraffe.profile.base.BaseViewModel
+import com.giraffe.profile.screens.profile.model.toUserUiModel
+import com.giraffe.profile.utils.Language
+import com.giraffe.profile.utils.LanguageHelper
+import com.giraffe.user.usecase.GetDarkModeUseCase
+import com.giraffe.user.usecase.GetLanguageUseCase
+import com.giraffe.user.usecase.GetUserUseCase
+import com.giraffe.user.usecase.IsLoggedInUseCase
+import com.giraffe.user.usecase.SetDarkModeUseCase
+import com.giraffe.user.usecase.SetLanguageUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import jakarta.inject.Inject
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+
+
+@HiltViewModel
+class SettingsViewModel @Inject constructor(
+    private val isLoggedInUseCase: IsLoggedInUseCase,
+    private val getUserUseCase: GetUserUseCase,
+    private val getDarkModeUseCase: GetDarkModeUseCase,
+    private val setDarkModeUseCase: SetDarkModeUseCase,
+    private val setLanguageUseCase: SetLanguageUseCase,
+    private val getLanguageUseCase: GetLanguageUseCase,
+) : BaseViewModel<SettingsScreenState, SettingsScreenEffect>(SettingsScreenState()),
+    SettingsInteractionListener {
+
+    init {
+        checkLoginStatus()
+        observeSettings()
+    }
+
+    private fun checkLoginStatus() {
+        viewModelScope.launch {
+            val isLoggedIn = isLoggedInUseCase()
+            updateState { it.copy(isLoggedIn = isLoggedIn) }
+            if (isLoggedIn) {
+                getUserProfile()
+            } else {
+                updateState { it.copy(isLoading = false) }
+            }
+        }
+    }
+
+    private fun observeSettings() {
+        getDarkModeUseCase().onEach { isDark ->
+            updateState { it.copy(isDarkMode = isDark) }
+        }.launchIn(viewModelScope)
+
+        getLanguageUseCase().onEach { langCode ->
+            val language = Language.entries.firstOrNull { it.code == langCode } ?: Language.ARABIC
+            updateState { it.copy(currentLanguage = language) }
+        }.launchIn(viewModelScope)
+    }
+
+    private fun getUserProfile() {
+        safeExecute(
+            onSuccess = { user ->
```

Comment:
to make this function clean sperate the success and error to function and put them like this:
```suggestion
            onSuccess = :: GetUserProfileSuccess
```
and same to Error